File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-2.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int n = nums.size(), l = 0, res = 0;

        for (int r = 0; r < n; ++r) {
            k -= nums[r] == 0;

            while (k < 0) {
                k += nums[l] == 0;
                ++l;
            }

            res = max(res, r - l + 1);
        }

        return res;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-2.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        n = len(nums)
        res, sol, total = [], [], 0

        def backtrack(i):
            nonlocal total
            if i == n or total > target:
                if total == target and sol not in res:
                    res.append(sol[:])
                return

            backtrack(i + 1)

            total += nums[i]
            sol.append(nums[i])

            backtrack(i)

            sol.pop()
            total -= nums[i]

        backtrack(0)
        return res